### PR TITLE
Convert TSDKCollectionObject identifer from NSInteger to NSString

### PR DESCRIPTION
**Pull request by SkylerSeamans**
Waiting for other branches to be merged.

### CODE REVIEW

File: TeamSnapSDK/SDK/CollectionJSON/TSDKCollectionObject.m [hunk 1]
```diff
@@ -499,8 +499,19 @@ + (BOOL)resolveInstanceMethod:(SEL)aSEL {
     }
 }
 
-- (NSInteger)objectIdentifier {
-    return [self getInteger:@"id"];
+- (NSString *)objectIdentifier {
+    if ((!_collection.data[@"id"]) || ([_collection.data[@"id"] isEqual:[NSNull null]])) {
+        return nil;
```
Comment (jrahaim) [inline TeamSnapSDK/SDK/CollectionJSON/TSDKCollectionObject.m]: Nonnull

File: TeamSnapSDK/SDK/DataTypes/Messages/TSDKMessage.m [hunk 2]
```diff
@@ -35,7 +35,7 @@ - (TSDKMessageType)messageTypeValue {
 
 + (void)actionMarkMessageAsRead:(TSDKMessage *)message completion:(TSDKCompletionBlock)completion {
     TSDKCollectionCommand *command = [[TSDKMessage commands] objectForKey:@"mark_message_as_read"];
-    command.data[@"id"] = [NSNumber numberWithInteger:message.objectIdentifier];
+    command.data[@"id"] = [NSNumber numberWithInteger:[message.objectIdentifier integerValue]];
```
Comment (jrahaim) [inline TeamSnapSDK/SDK/DataTypes/Messages/TSDKMessage.m]: shouldn't this be command.data[@"id"] = message.objectIdentifier?

Comment (SkylerSeamans) [inline TeamSnapSDK/SDK/DataTypes/Messages/TSDKMessage.m]: I'm not sure if the API would accept "3" versus 3.

**Comment by jrahaim** [comment]
TSDKMember.m Line 105
```[poisonPill setInteger:self.objectIdentifier forKey:@"id"];```

**Comment by jrahaim** [comment]
TSDKMessage.m
line 38
```command.data[@"id"] = [NSNumber numberWithInteger:message.objectIdentifier];```

lines 53 & 55
```
[commaSeparatedIds appendString:[NSString stringWithFormat:@"%ld", (long)message.objectIdentifier]];
```

**Comment by jrahaim** [comment]
I am not sure why this doesn't trigger a warning. 

TSDKUser -> teamIdsForAllMyTeamsWithConfiguration
```                
NSString *teamId = [NSString stringWithFormat:@"%ld", (long)member.teamId];
                if (teamId != 0) {
                    if (![teamIds containsObject:teamId]) {
                        [teamIds addObject:[NSString stringWithFormat:@"%ld", (long)[teamId integerValue]]];
                    }
                }
```

**Pull request merged**